### PR TITLE
fix: vault debtRatio update on _reportLoss

### LINKED ISSUE #184: When strategy has loss, call updateStrategyDebtRatio method to update strategies' debtRatio after harvest will  revert.
It's because that _reportLoss only adjust the strategy's debtRatio but not change the vault's debtRatio. it will cause updateStrategyDebtRatio execute failed.

### PR DESCRIPTION
**Pull request by steffenix**
fixes #184

### CODE REVIEW

File: contracts/Vault.vy [hunk 1]
```diff
@@ -1482,8 +1482,9 @@ def _reportLoss(strategy: address, loss: uint256):
 
     # Also, make sure we reduce our trust with the strategy by the same amount
     debtRatio: uint256 = self.strategies[strategy].debtRatio
-    self.strategies[strategy].debtRatio -= min(loss * MAX_BPS / self._totalAssets(), debtRatio)
-
+    ratio_change: uint256 = min(loss * MAX_BPS / self._totalAssets(), debtRatio)
+    self.strategies[strategy].debtRatio -= ratio_change 
+    self.debtRatio -= ratio_change
```
Comment (steffenix) [inline contracts/Vault.vy:1487]: Could also be done with:

```
    self.debtRatio -= self.strategies[strategy].debtRatio
    self.strategies[strategy].debtRatio -= ratio_change 
    self.debtRatio += self.strategies[strategy].debtRatio
```

**Review by fubuloubu — APPROVED** [review]
Add a check in a test case that shows the ratio change is handled appropiately

Bonus points if you add the test case first and show it fails lol

File: contracts/Vault.vy (hunk 1 above)
Comment (fubuloubu) [reply contracts/Vault.vy:1487]: Naw, I like way it's handled in how you did it already

Comment (steffenix) [reply contracts/Vault.vy:1487]: Should I change?

Comment (steffenix) [reply contracts/Vault.vy:1487]: ok

**Comment by steffenix** [comment]
> Add a check in a test case that shows the ratio change is handled appropiately
> 
> Bonus points if you add the test case first and show it fails lol

Couldn't find a test but was planning to add one tomorrow. Will make sure it fails on master.

**Comment by steffenix** [comment]
@fubuloubu The failing test
https://github.com/iearn-finance/yearn-vaults/pull/191

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
Awesome! I'm gonna merge this one, and close the other one. Next time you could just use `git rebase -i master` to rearrange the commits, force push just the test first (showing it fails), and then adding your fix afterwards (showing it passes)

Awesome work!

File: tests/functional/vault/test_misc.py [hunk 2]
```diff
@@ -145,3 +145,16 @@ def test_deposit_withdraw_faillure(token, gov, vault):
 
     with brownie.reverts():
         vault.withdraw(vault.balanceOf(gov), {"from": gov})
+
+
+def test_report_loss(token, gov, vault, strategy, accounts):
+    strategy.harvest()
+    strategy_account = accounts.at(strategy.address, force=True)
+    token.transfer(gov, token.balanceOf(strategy.address), {"from": strategy_account})
+    assert token.balanceOf(strategy.address) == 0
```
Comment (fubuloubu) [inline tests/functional/vault/test_misc.py:154]: We added `_takeFunds` for this purpose to `TestStrategy`:
https://github.com/iearn-finance/yearn-vaults/blob/6ba0909826f7eb5a9671b9c047687c8b519f0e77/contracts/test/TestStrategy.sol#L27-L29

File: tests/functional/vault/test_misc.py [hunk 3]
```diff
@@ -145,3 +145,16 @@ def test_deposit_withdraw_faillure(token, gov, vault):
...
+
+    # Make sure we do not send more funds to the strategy.
+    strategy.harvest()
+    assert token.balanceOf(strategy.address) == 0
```
Comment (fubuloubu) [inline tests/functional/vault/test_misc.py:158]: BTW this works in Brownie:
```suggestion
    assert token.balanceOf(strategy) == 0
```

File: tests/functional/vault/test_misc.py (hunk 2 above)
Comment (steffenix) [reply tests/functional/vault/test_misc.py:154]: Oh, got it!

**Comment by steffenix** [comment]
> Awesome! I'm gonna merge this one, and close the other one. Next time you could just use `git rebase -i master` to rearrange the commits, force push just the test first (showing it fails), and then adding your fix afterwards (showing it passes)
> 
> Awesome work!

Didn't thought about only pushing only the first commit, great idea.

**Pull request merged**